src/app/modules/verification/verification.model.ts

```diff
@@ -134,8 +132,10 @@ const compileVerificationModel = (db: Mongoose): IVerificationModel => {
   ): Promise<IVerificationSchema | null> {
     const formFields = getVerifiableFormFields(form.form_fields)
     if (form.responseMode === FormResponseMode.Encrypt) {
-      const { payments_field } = form as IEncryptedFormSchema
-      const paymentField = getVerifiablePaymentContactField(payments_field)
+      const { payments_channel } = form as IEncryptedFormSchema
+      const paymentField = getVerifiablePaymentContactField(
+        payments_channel.channel !== PaymentChannel.Unconnected,
```

Comment:
Same comment as earlier: is this test enough to guarantee that payment is enabled on the form?

Reply:
Hm okay true, description and amount will be a stronger guarantee for this. Will do

Reply:
Done in [d1ba976](https://github.com/opengovsg/FormSG/pull/6398/commits/d1ba976a4491d34b39b417d43557f36b865fe73b)